src/app/clusters/level-control/level-control.cpp

```diff
@@ -960,18 +960,56 @@ static void moveHandler(app::CommandHandler * commandObj, const app::ConcreteCom
         goto send_default_response;
     }
 
+    // Always validate the rate parameter received. A rate of 0 is invalid.
     if (!rate.IsNull() && (rate.Value() == 0))
     {
-        // Move at a rate of zero is no move at all. Immediately succeed without touching anything.
-        ChipLogProgress(Zcl, "Immediate success due to move rate of 0 (would move at no rate).");
-        status = Status::Success;
+        // Providing rate of 0 is not allowed.
+        status = Status::InvalidCommand;
         goto send_default_response;
     }
 
+    uint8_t eventDuration; // use this local var so state->eventDurationMs is only set once the command is validated.
+#ifndef IGNORE_LEVEL_CONTROL_CLUSTER_TRANSITION
+    // If the Rate field is null, the device should move at the default move rate, if available,
+    // Otherwise, move as fast as possible
+    if (rate.IsNull())
+    {
+        app::DataModel::Nullable<uint8_t> defaultMoveRate;
+        status = Attributes::DefaultMoveRate::Get(endpoint, defaultMoveRate);
+        if (status != Status::Success || defaultMoveRate.IsNull())
+        {
+            ChipLogProgress(Zcl, "ERR: reading default move rate %x", to_underlying(status));
+            eventDuration = FASTEST_TRANSITION_TIME_MS;
+        }
+        else
+        {
+            // Error out if DefaultMoveRate is equal to 0
```

Comment:
The spec has no provisions for this; another question that was raised on the spec PR.

Reply:
You are correct, the spec must clarify this. This implementation comes from the Slack conversation where this was the agreed behaviour.

I'll wait for spec to change or merge this

Reply:
So the spec pr changes the min value for DefaultMoveRate to 1. I updated the level control XML with this change. Going forward, it will not be possible to set a DefaultMoveRate of 0.

On the off chance that someone configured the DefaultMoveRate to 0 with a previous matter version, this code would return an invalid command.

I am not sure if I should leave this behaviour as is, since 0 is an invalid value for the attribute, or rather move at the fastest rate possible as if it was null since it was previously possible to set it to 0

Reply:
That sounds like a spec question, really; they should talk about the backwards compat behavior in there.